template/node/Dockerfile

```diff
@@ -12,27 +12,31 @@ WORKDIR /root/
 # Turn down the verbosity to default level.
 ENV NPM_CONFIG_LOGLEVEL warn
 
+RUN mkdir -p /usr/local/app/function
+
 # Wrapper/boot-strapper
-COPY package.json       .
+COPY package.json       /usr/local/app/
+
+WORKDIR /usr/local/app
 RUN npm i
 
 # Function
-COPY index.js           .
-RUN mkdir -p ./root/function
+COPY index.js           /usr/local/app/
 
-COPY function/*.json    ./function/
-WORKDIR /root/function
+COPY function/*.json    /usr/local/app/function/
+WORKDIR /usr/local/app/function
 RUN npm i || :
-WORKDIR /root/
-COPY function           function
-WORKDIR /root/function
+WORKDIR /usr/local/app/
+COPY function           ./function
 
-WORKDIR /root/
+WORKDIR /usr/local/app/
+
+USER node
```

Comment:
Great start! Maybe the username could be `app` and we could use that across the board - like a copy/paste segment for all the templates?

Reply:
that makes the most sense. I'll work on it this evening

Reply:
Just picking up on the user, and acknowledging that its largely academic, but why dont we use `faas` or `openfaas`?  Setting this as a sort of principle might be helpful further down the road, when 3rd party templates start to emerge.

Reply:
I'd just keep it generic like `app` for now. Always good to think forward though so thanks for your input.